src/stratis_cli/_actions/_data.py

```diff
@@ -207,29 +209,71 @@
 DBUS_TIMEOUT_SECONDS = 120
 
 
+def _get_timeout(value):
+    """
+    Turn an input str or int into a float timeout value.
+
+    :param value: the input str or int
+    :type value: str or int
+    :raises StratisCliEnvironmentError:
+    :returns: float
+    """
+    # Ensure the input str is not a float
+    if isinstance(value, float):
+        raise StratisCliEnvironmentError(
+            "The timeout value you provided is a float; it should be an integer."
+        )
+
+    try:
+        timeout_int = int(value)
+
+    except ValueError:
+        raise StratisCliEnvironmentError(
+            "The timeout value you provided is not an integer."
+        )
+
+    # Ensure the integer is not too small
+    if timeout_int < -1:
+        raise StratisCliEnvironmentError(
+            "The timeout value you provided is smaller than the smallest acceptable value, -1."
+        )
+
+    # Ensure the integer is not too large
+    if timeout_int > (1 << 30) - 1:
```

Comment:
Make this value a constant, and call it something like ```MAXIMUM_DBUS_TIMEOUT_MS```. You can put the definition next to the ```DBUS_TIMEOUT_SECONDS``` definition. You should use the constant value in the error  message, also.